src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -637,11 +634,9 @@ function discoverController(
     $scope.minimumVisibleRows = $scope.hits;
   };
 
-  let loadingVis;
   function setupVisualization() {
-    // If we're not setting anything up we need to return an empty promise
-    if (!$scope.opts.timefield) return Promise.resolve();
-    if (loadingVis) return loadingVis;
+    // If no timefield has been specified we don't create a histogram of messages
+    if (!$scope.opts.timefield) return;
```

Comment:
Isn't this redundant with line 391? How about removing this one if that is the case?

Reply:
We are calling that method also from line 434 when updating the data source, so I would rather leave that additional check in there.

Reply:
Yes, you're right. Interestingly, `updateTime` is called unconditionally in [line 431](https://github.com/timroes/kibana/blob/5ef9b6c4e0f4546935473ccf3704a3f43a6fe8b1/src/core_plugins/kibana/public/discover/controllers/discover.js#L431). 🤔

Anyway, don't let this block you 😉